docs/source/examples/syncing-code-artifacts.rst

```diff
@@ -18,49 +18,111 @@ Here, "upload" means uploading files from your local machine (or a cloud object
 
-(1) by the :code:`workdir` field in a :ref:`task YAML file <yaml-spec>`, or
-(2) by the command line option :code:`--workdir`:
+.. tab-set::
 
-.. code-block:: console
+    .. tab-item:: Local Directory
+        :sync: local-directory-tab
 
-  $ # Assuming task.yaml has a 'workdir: <path>' field, these commands
-  $ # sync the workdir to the cluster:
-  $ sky launch -c dev task.yaml
-  $ sky exec dev task.yaml
+        If ``workdir`` is a local path, the entire directory is synced to the remote cluster. To exclude files from syncing, see :ref:`exclude-uploading-files`.
 
-  $ # Add a --workdir flag if the yaml doesn't contain the field, or
-  $ # to override it.
+        If a relative path is used, it's evaluated relative to the location from which ``sky`` is called.
 
-These commands sync the working directory to :code:`~/sky_workdir` on the remote
-VMs.  The task is invoked under that working directory (so that it can call
-scripts, access checkpoints, etc.).
+        The working directory can be configured either
 
-.. note::
-  To exclude large files from being uploaded, see :ref:`exclude-uploading-files`.
+        1. by the :code:`workdir` field in a :ref:`task YAML file <yaml-spec-workdir>`, or
+        2. by the command line option :code:`--workdir`:
 
-.. note::
+        .. code-block:: console
+
+          $ # Assuming task.yaml has a 'workdir: <path>' field
+          $ # workdir: ~/my-task-code
+
+          $ # Sync the workdir to the ~/sky_workdir of the cluster:
+          $ sky launch -c dev task.yaml
+          $ sky exec dev task.yaml
+
+          $ # Add a --workdir flag if the yaml doesn't contain the field, or
+          $ # to override it.
+
+        .. note::
+          To exclude large files from being uploaded, see :ref:`exclude-uploading-files`.
+
+        .. note::
+
+          You can keep and edit code in one central place---the local machine where
+          :code:`sky` is used---and have them transparently synced to multiple remote
+          clusters for execution:
+
+          .. code-block:: console
+
+            $ sky exec cluster0 task.yaml
+
+            $ # Make local edits to the workdir...
+            $ # cluster1 will get the updated code.
+            $ sky exec cluster1 task.yaml
+
+    .. tab-item:: Git Repository
+        :sync: git-repository-tab
+
+        If ``workdir`` is a git repository, the ``workdir.url`` field is required and can be in one of the following formats:
+
+        * HTTPS: ``https://github.com/skypilot-org/skypilot.git``
+        * SSH: ``ssh://git@github.com/skypilot-org/skypilot.git``
+        * SCP: ``git@github.com:skypilot-org/skypilot.git``
+
+        The ``workdir.ref`` field specifies the git reference to checkout, which can be:
```

Comment:
Can we add a one or two most common snippets here, so that users can easily copy paste? (This is appropriate for user-facing guide, rather than presenting reference-level details. Some of these details can move back to YAML spec and we refer to them from here.)

Reply:
Updated